src/models.py

```diff
@@ -677,7 +684,7 @@ def __init__(self, args, sent_encoder, vocab):
         self.use_bert = bool(args.bert_model_name)
         self.sep_embs_for_skip = args.sep_embs_for_skip
 
-    def forward(self, task, batch, predict=False):
+    def forward(self, task, batch, cuda_device, predict=False):
```

Comment:
I'm still confused as to why you need to pass in the cuda device now whereas for the other tasks we didn't need to. Could you explain a bit more?